file.go

```diff
@@ -460,18 +461,24 @@ func (c *fileColumnChunk) Pages() Pages {
 	return r
 }
 
-func (c *fileColumnChunk) ColumnIndex() ColumnIndex {
+func (c *fileColumnChunk) ColumnIndex() (ColumnIndex, error) {
+	if err := c.readColumnIndex(); err != nil {
+		return nil, err
+	}
 	if c.columnIndex == nil {
-		return nil
+		return nil, nil
```

Comment:
I know this was the historical behavior, but I imagine this might be a bit unintuitive now, after verifying that a method didn't return an error, we usually expect the returned value to be usable. 

What do you think about adding a global error value like ErrMissingColumnIndex that we would return in this case? Since we are returning an error now, we could better work with the code that Go developers are used to writing (doing error handling with errors.Is, etc...).

Reply:
Happy to make a follow up for this.

Reply:
Thanks, that will be really useful 👍